private/pkg/transport/http/httpclient/httpclient.go

```diff
@@ -16,65 +16,62 @@ package httpclient
 
 import (
 	"crypto/tls"
-	"fmt"
-	"io"
 	"net/http"
 	"net/url"
-
-	"go.uber.org/multierr"
 )
 
-// Client is a client.
-type Client interface {
-	// Do matches http.Client.
-	//
-	// This allows Client to be dropped in for http.Client.
-	Do(request *http.Request) (*http.Response, error)
-	// ParseAddress parses the given address.
-	//
-	// If the address has a scheme, this is a no-op.
-	// If the address does not have a scheme, this adds https:// if TLS was configured,
-	// and http:// if TLS was not configured.
-	ParseAddress(address string) string
-	// Transport returns the http.RoundTripper configured on
-	// this client.
-	Transport() http.RoundTripper
-}
-
 // NewClient returns a new Client.
-func NewClient(options ...ClientOption) Client {
+func NewClient(options ...ClientOption) *http.Client {
 	return newClient(options...)
 }
 
 // ClientOption is an option for a new Client.
-type ClientOption func(*client)
+type ClientOption func(*clientOptions)
+
+// ClientInterceptorFunc is a function that wraps a RoundTripper with any interceptors
+type ClientInterceptorFunc func(http.RoundTripper) http.RoundTripper
 
-// ClientWithTLSConfig returns a new ClientOption to use the tls.Config.
+// WithTLSConfig returns a new ClientOption to use the tls.Config.
 //
 // The default is to use no TLS.
-func ClientWithTLSConfig(tlsConfig *tls.Config) ClientOption {
-	return func(client *client) {
-		client.tlsConfig = tlsConfig
+func WithTLSConfig(tlsConfig *tls.Config) ClientOption {
+	return func(opts *clientOptions) {
+		opts.tlsConfig = tlsConfig
 	}
 }
 
-// ClientWithObservability returns a new ClientOption to use
+// WithObservability returns a new ClientOption to use
 // OpenCensus tracing and metrics.
 //
 // The default is to use no observability.
-func ClientWithObservability() ClientOption {
-	return func(client *client) {
-		client.observability = true
+func WithObservability() ClientOption {
+	return func(opts *clientOptions) {
+		opts.observability = true
+	}
+}
```

Comment:
I know the stutter is a little frustrating, but we might actually want to leave these options as-is with the `Client*` prefix. It's consistent with what you see in other packages like `bufmodule.ModuleWithTargetPaths`.